src/Uno.UI.Tests/Windows_UI_Xaml/Given_ThemeResource.cs

```diff
@@ -40,8 +49,160 @@ public void When_System_ThemeResource_Dark()
+			app.HostView.Children.Add(control);
+
+			Assert.AreEqual(ApplicationTheme.Light, app.RequestedTheme);
+			Assert.AreEqual(Colors.RosyBrown, (control.TestBorder.BorderBrush as SolidColorBrush).Color);
+		}
+
+		[TestMethod]
+		public void When_Inherited_In_Template_Applied()
+		{
+			var control = new Test_Control();
+
+			control.InlineTemplateControl.ApplyTemplate();
+			control.TemplateFromResourceControl.ApplyTemplate();
+
+			var text2InlineBefore = control.InlineTemplateControl.TextBlock2.Text;
+			Assert.AreEqual("LocalVisualTree", text2InlineBefore);
+			var text2ResourceTemplateBefore = control.TemplateFromResourceControl.TextBlock2.Text;
+			Assert.AreEqual("OuterVisualTree", text2ResourceTemplateBefore);
+
+			var text4InlineBefore = control.InlineTemplateControl.TextBlock4.Text;
+			Assert.AreEqual("ApplicationLevel", text4InlineBefore);
+			var text4ResourceTemplateBefore = control.TemplateFromResourceControl.TextBlock4.Text;
+			Assert.AreEqual("ApplicationLevel", text4ResourceTemplateBefore);
+		}
+
+		[TestMethod]
+		public void When_Inherited_In_Template_Applied_XAML_Scope()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+
+			control.InlineTemplateControl.ApplyTemplate();
+			control.TemplateFromResourceControl.ApplyTemplate();
+
+			var text2InlineBefore = control.InlineTemplateControl.TextBlock2.Text;
+			var text2ResourceTemplateBefore = control.TemplateFromResourceControl.TextBlock2.Text;
+
+			var text4InlineBefore = control.InlineTemplateControl.TextBlock4.Text;
+			var text4ResourceTemplateBefore = control.TemplateFromResourceControl.TextBlock4.Text;
+
+			app.HostView.Children.Add(control);
+
+			var text2InlineAfter = control.InlineTemplateControl.TextBlock2.Text;
+			var text2ResourceTemplateAfter = control.TemplateFromResourceControl.TextBlock2.Text;
+
+			Assert.AreEqual("LocalVisualTree", text2InlineBefore);
+			Assert.AreEqual("OuterVisualTree", text2ResourceTemplateBefore);
+
+			Assert.AreEqual("ApplicationLevel", text4InlineBefore);
+			Assert.AreEqual("ApplicationLevel", text4ResourceTemplateBefore);
+
+			Assert.AreEqual("LocalVisualTree", text2InlineAfter);
+			Assert.AreEqual("LocalVisualTree", text2ResourceTemplateAfter);
+		}
+
+		[TestMethod]
+		public async Task When_Theme_Changed()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+			app.HostView.Children.Add(control);
+
+			control.Measure(new Size(1000, 1000));
+
+			var textLightThemeMarkup = control.TemplateFromResourceControl.TextBlock6.Text;
+
+			Assert.AreEqual("LocalVisualTreeLight", textLightThemeMarkup);
+
+			try
+			{
+				if (await SwapSystemTheme())
+				{
+					var textDarkThemeMarkup = control.TemplateFromResourceControl.TextBlock6.Text;
+					Assert.AreEqual("LocalVisualTreeDark", textDarkThemeMarkup); //ThemeResource markup change lookup uses the visual tree (rather than original XAML namescope)
+					;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)